runtime/jvmti/jvmtiStartup.c

```diff
@@ -1061,6 +1066,171 @@ createXrunLibraries(J9JavaVM * vm)
 	return JNI_OK;
 }
 
+#if defined(AIXPPC)
+/**
+ * Check if the first dirLen bytes of the file name is a library path.
+ *
+ * @param[in] fileName the file name containing a path
+ * @param[in] dirLen the number of bytes to be compared
+ * @param[in] libpath the library path returned from loadquery(L_GETLIBPATH)
+ *
+ * @return TRUE if the first dirLen bytes of the file name is a library path, FALSE otherwise
+ */
+static BOOLEAN
+isFileNameInLibPath(const char *fileName, UDATA dirLen, char *libpath)
+{
+	BOOLEAN result = FALSE;
+	char *path = NULL;
+	char *strTokPtr = NULL;
+
+	path = strtok_r(libpath, ":", &strTokPtr);
+	while (NULL != path) {
+		if (0 != strncmp(fileName, path, dirLen)) {
+			/* continue to search the libpath */
+			Trc_JVMTI_isFileNameInLibPath_path_not_match(fileName, path, dirLen);
+			path = strtok_r(NULL, ":", &strTokPtr);
+		} else {
+			/* a match found */
+			result = TRUE;
+			Trc_JVMTI_isFileNameInLibPath_path_match(fileName, path, dirLen);
+			break;
+		}
```

Comment:
This can yield the wrong answer, for example when `fileName = "/u/jo/libfoo.so"` and `LIBPATH = "/u/jose:/u/joseph"` because it only considers the first 5 characters. After adding the missing condition, suggest inverting to get:
```c
		if ((0 == strncmp(fileName, path, dirLen) && ('\0' == path[dirLen])) {
			/* a match found */
			result = TRUE;
			Trc_JVMTI_isFileNameInLibPath_path_match(fileName, path, dirLen);
			break;
		}
		/* continue to search the libpath */
		Trc_JVMTI_isFileNameInLibPath_path_not_match(fileName, path, dirLen);
```

Reply:
Fixed